src/R/Interpreters/Impl/Implementation/RInstallation.cs

```diff
@@ -82,13 +62,22 @@ public sealed class RInstallation {
         private IEnumerable<IRInterpreterInfo> GetCompatibleEnginesFromRegistry(ISupportedRVersionRange svr) {
             svr = svr ?? new SupportedRVersionRange();
             var engines = GetInstalledEnginesFromRegistry().Where(e => svr.IsCompatibleVersion(e.Version));
-            // Remove duplicates (MRC registers under multiple keys)
-            var mrc = engines.FirstOrDefault(e => e.Name.Contains("Microsoft"));
-            if(mrc != null) {
-                var dupes = engines.Where(e => e.InstallPath.EqualsIgnoreCase(mrc.InstallPath)).Except(new IRInterpreterInfo[] { mrc });
-                engines = engines.Except(dupes);
+            // Among duplicates by path (MRC registers under multiple keys) take the highest version
+            var microsoftEngines = engines.Where(e => e.Name.Contains("Microsoft") || e.InstallPath.Contains("Microsoft"));
+            var installLocations = microsoftEngines.Select(e => e.InstallPath.TrimTrailingSlash()).Distinct();
```

Comment:
I think what you want to do here is group by path, then take highest in each group. Probably easier to write as a query; something like this:
```cs
var microsoftEngines =
  from e in engines
  where e.Name.Contains("Microsoft") || e.InstallPath.Contains("Microsoft")
  group e by e.InstallPath.TrimTrailingSlash() into g
  select g.OrderByDescending(e => e.Version).First();
```
Also, I think you might as well dedup all keys, not just MRC. If they never duplicate, it doesn't really do anything. But in case they do, this would take care of them all.